apps/web/server/routers/viewer/eventTypes.tsx

```diff
@@ -128,6 +128,17 @@ export const eventTypesRouter = createProtectedRouter()
         data.locations = [{ type: "integrations:daily" }];
       }
 
+      //Check if given user already has slug for their eventTypes
+      const userEventExists = await ctx.prisma.eventType.findFirst({
+        where: {
+          userId,
+          slug: data.slug,
+        },
+      });
+      if (userEventExists) {
+        throw new TRPCError({ code: "BAD_REQUEST", message: "URL Slug already exists for given user." });
+      }
```

Comment:
Instead of making a new query why don't we catch and parse the "unique constrain" error that prisma already throws? 

```
@calcom/web:dev:   originalError: PrismaClientKnownRequestError:
@calcom/web:dev:   Invalid `prisma.eventType.create()` invocation:
@calcom/web:dev:
@calcom/web:dev:
@calcom/web:dev:     Unique constraint failed on the fields: (`userId`,`slug`)
```

Reply:
Refactored in [b3c952b](https://github.com/calcom/cal.com/pull/2757/commits/b3c952b71e3d5c09e22a2932592e3233191fff4e)